test/Chainweb/Test/Pact/DbCacheTest.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Chainweb.Test.Pact.DbCacheTest (tests) where
+
+import Chainweb.Pact.Backend.DbCache
+
+import Control.Monad.State.Strict
+
+import Data.Aeson
+import Data.ByteString.Lazy (toStrict)
+
+import Database.SQLite3.Direct
+
+import GHC.Compact
+
+import Test.Tasty
+import Test.Tasty.HUnit
+
+tests :: TestTree
+tests = testGroup "Chainweb.Test.Pact.DbCacheTest"
+    [ testCache ]
+
+defaultChunkSize :: Int
+defaultChunkSize = 32768
+
+-- | Create entries with a size the corresponds to the default
+-- chunk size of the cache.
+--
+entry :: MonadIO m => Int -> Char -> m ([String], Int)
+entry i c = do
+    let e = [replicate (i * defaultChunkSize `div` 256) c]
+    s <- liftIO $ compactSize =<< compact e
+    return (e, fromIntegral s)
+
+combineEntries :: MonadIO m => Monoid a => a -> a -> m (a, Int)
```

Comment:
I don't think we need this. The original version was using string literals so `<>` was useful, but now `entry` is explicitly sized in integers so let's just call it directly, such that if `a` is `1`, `b` is `2` and `c` is `4`, etc.